pkg/daemon/daemon.go

```diff
@@ -2889,3 +2890,43 @@ func (dn *Daemon) getUnsupportedPackages() {
 	}
 	unsupportedPackages.WithLabelValues(dn.name).Set(float64(unsupportedPackageCount))
 }
+
+func (dn *Daemon) preserveDaemonLogs() {
+	// Find current daemon log location
+	var currentLogLocation, logDir string
+	if walkErr := filepath.WalkDir("/var/log/pods/", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.Contains(d.Name(), "openshift-machine-config-operator_machine-config-daemon-") {
+			currentLogLocation = path
+			logDir = d.Name()
+		}
+		return nil
+	}); walkErr != nil {
+		klog.Errorf("Daemon logs could could not be found due to error: %v", walkErr)
+		return
+	}
+
+	baseDir := "/etc/machine-config-daemon/previous-logs/"
+	// Remove the old logs if it exists
+	if err := os.RemoveAll(baseDir); err != nil {
```

Comment:
I opted to remove any old logs as I was a bit worried that we might fill up the disk with every daemon restart. I also figured the (n-1) set of logs are most critical, but I'm open to other approaches here too :thinking: